docs/site/Loopback-component-authorization.md

```diff
@@ -8,543 +8,227 @@ permalink: /doc/en/lb4/Loopback-component-authorization.html
-  over and above role permissions.
-- **_UserPermissionsFn_** - This is going to be the interface for user
-  permissions manipulation, if required any.
+- an interceptor to enforce authorization with
+  `user provided voters/authorizers`
+  ([see, Programming Access Policies](##Programming-Access-Policies)) on API
+  calls using,
+  - authorization metadata added by the decorator on the target controller
+    method (roles, scopes, voters)
+  - a Principal/Subject deducted from the incoming request
+  - [a decision matrix](##Authorization-by-decision-matrix)
 
-The PermissionKey is an enum containing all possible permission keys. Here is a
-sample.
+![Authorization](./imgs/authorization.png)
 
-{% include code-caption.html content="/src/authorization/permission-key.ts" %}
+## Authorization Component
 
-```ts
-export const enum PermissionKey {
-  // For accessing own (logged in user) profile
-  ViewOwnUser = 'ViewOwnUser',
-  // For accessing other users profile.
-  ViewAnyUser = 'ViewAnyUser',
-  // For creating a user
-  CreateAnyUser = 'CreateAnyUser',
-  // For updating own (logged in user) profile
-  UpdateOwnUser = 'UpdateOwnUser',
-  // For updating other users profile
-  UpdateAnyUser = 'UpdateAnyUser',
-  // For deleting a user
-  DeleteAnyUser = 'DeleteAnyUser',
-
-  // For accessing a role
-  ViewRoles = 'ViewRoles',
-  // For creating a role
-  CreateRoles = 'CreateRoles',
-  // For updating a role info
-  UpdateRoles = 'UpdateRoles',
-  // For removing a role
-  DeleteRoles = 'DeleteRoles',
-}
-```
+- The `@loopback/authorization` package exports an
+  [Authorization Component class](https://github.com/strongloop/loopback-next/blob/master/packages/authorization/src/authorization-component.ts).
 
-Next, we create the binding keys for each type and accessor key for method
-decorator.
+  - Developers will have to register this component to use access control
+    features in their application.
 
-{% include code-caption.html content="/src/authorization/keys.ts" %}
+  - The component binds an in-built interceptor (`Authorization Interceptor`) to
+    all API calls.
 
-```ts
-import {BindingKey} from '@loopback/context';
-import {MetadataAccessor} from '@loopback/metadata';
-import {AuthorizeFn, AuthorizationMetadata, UserPermissionsFn} from './types';
-
-/**
- * Binding keys used by this component.
- */
-export namespace AuthorizationBindings {
-  export const AUTHORIZE_ACTION = BindingKey.create<AuthorizeFn>(
-    'userAuthorization.actions.authorize',
-  );
-
-  export const METADATA = BindingKey.create<AuthorizationMetadata | undefined>(
-    'userAuthorization.operationMetadata',
-  );
-
-  export const USER_PERMISSIONS = BindingKey.create<UserPermissionsFn>(
-    'userAuthorization.actions.userPermissions',
-  );
-}
+    - The interceptor checks to see if the endpoint is annotated with an
+      authorization specification.
+    - It executes all classes tagged as `Authorizer` to enforce access/privilege
```

Comment:
The `authorizer` could be a function, `Authorizer` is a type, see https://github.com/strongloop/loopback-next/blob/master/packages/authorization/src/types.ts#L100

Reply:
And interceptor also collect `voters` provided in the `@authorize()` decorator per endpoint.

Reply:
agreed